pkg/controller/common/helpers.go

```diff
@@ -236,13 +236,53 @@ func ConvertRawExtIgnitionToV2(inRawExtIgn *runtime.RawExtension) (runtime.RawEx
 	return outRawExt, nil
 }
 
+// Ensures SSH keys are unique for a given Ign 2 PasswdUser
+func dedupePasswdUserSSHKeys(passwdUser ign2types.PasswdUser) ign2types.PasswdUser {
+	// Map for checking for duplicates.
+	knownSSHKeys := map[ign2types.SSHAuthorizedKey]bool{}
+
+	// Preserve ordering of SSH keys.
+	dedupedSSHKeys := []ign2types.SSHAuthorizedKey{}
+
+	for _, sshKey := range passwdUser.SSHAuthorizedKeys {
+		if _, isKnown := knownSSHKeys[sshKey]; isKnown {
+			// We've seen this key before warn and move on.
+			glog.Warningf("duplicate SSH public key found: %s", sshKey)
+			continue
+		}
+
+		// We haven't seen this key before, add it.
+		dedupedSSHKeys = append(dedupedSSHKeys, sshKey)
+		knownSSHKeys[sshKey] = true
+	}
+
+	// Overwrite the keys with the deduped list.
+	passwdUser.SSHAuthorizedKeys = dedupedSSHKeys
+
+	return passwdUser
+}
+
+// Ensures SSH keys are unique for all Ign 2 PasswdUsers in a slice
+// See: https://bugzilla.redhat.com/show_bug.cgi?id=1934176
+func dedupeSSHKeys(passwdUsers []ign2types.PasswdUser) []ign2types.PasswdUser {
```

Comment:
Given that the below function isn't really needed, I think we can also remove this function, and just have one `dedupePasswdUserSSHKeys`

Reply:
Fine with me!